kong/db/schema/entities/routes.lua

```diff
@@ -1,25 +1,92 @@
 local typedefs = require("kong.db.schema.typedefs")
 local deprecation = require("kong.deprecation")
 
+
 local kong_router_flavor = kong and kong.configuration and kong.configuration.router_flavor
 
+
+local PATH_V1_DEPRECATION_MSG =
+  "path_handling='v1' is deprecated and " ..
+  (kong_router_flavor == "traditional" and
+    "will be removed in future version, " or
+    "will not work under 'expressions' or 'traditional_compatible' router_flavor, ") ..
+  "please use path_handling='v0' instead"
+
+
+local entity_checks = {
+  { conditional = { if_field = "protocols",
+                    if_match = { elements = { type = "string", not_one_of = { "grpcs", "https", "tls", "tls_passthrough" }}},
+                    then_field = "snis",
+                    then_match = { len_eq = 0 },
+                    then_err = "'snis' can only be set when 'protocols' is 'grpcs', 'https', 'tls' or 'tls_passthrough'",
+                  }
+  },
+
+  { custom_entity_check = {
+    field_sources = { "path_handling" },
+    fn = function(entity)
+      if entity.path_handling == "v1" then
+        deprecation(PATH_V1_DEPRECATION_MSG, { after = "3.0", })
+      end
+
+      return true
+    end,
+  }},
+}
+
+
 -- works with both `traditional_compatible` and `expressions` routes
 local validate_route
-if kong_router_flavor ~= "traditional" then
+if kong_router_flavor == "traditional_compatible" or kong_router_flavor == "expressions" then
   local ipairs = ipairs
   local tonumber = tonumber
   local re_match = ngx.re.match
 
+  local MAX_EXPR_PRIORITY = 2^56
```

Comment:
```suggestion
  local MAX_EXPR_PRIORITY = 2^52
```

I think LuaJIT can not support numbers larger than `2^52` without losing precision, so `2^52` may be a more reasonable limit here.

Also, maybe we can use something like https://github.com/Kong/kong/blob/1714e26e033ee7b0d296739149c8b19e2d96dc0e/kong/db/schema/entities/targets.lua#L32 and declaratively, I know the @Kong/koko-engineering team would like declarative checks better than imperative ones.

Reply:
On the Koko side we're setting the limit to `2^53-1`
https://github.com/Kong/koko-private/blob/f26c9f3fec7620eb1d5c331424efc743a9c12030/internal/resource/route.go#L51

Reply:
done, using between 0 to 2^53 -1.